Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationResultDispatchingValidator.cs

```diff
@@ -36,16 +36,71 @@ public void DispatchValidationFailures (IBusinessObjectValidationResult validati
       ArgumentUtility.CheckNotNull("validationResult", validationResult);
 
       var bocListControl = GetControlToValidate();
+      var dispatchToValidationFailureRepository = bocListControl
+          .GetColumnDefinitions()
+          .Any(cd => cd is BocValidationErrorIndicatorColumnDefinition);
 
+      if (dispatchToValidationFailureRepository)
+        DispatchToValidationFailureRepository(validationResult, bocListControl);
+      else
+        DispatchToValidationResultDispatchers(validationResult, bocListControl);
+    }
+
+    private void DispatchToValidationResultDispatchers (IBusinessObjectValidationResult validationResult, BocList bocListControl)
+    {
       var validatorsMatchingToControls = EnumerableUtility.SelectRecursiveDepthFirst(
               bocListControl as Control,
-              child => child.Controls.Cast<Control>().Where(item => !(item is INamingContainer)))
+              child => child.Controls.Cast<Control>().Where(item => item is not INamingContainer))
           .OfType<IBusinessObjectBoundEditableWebControlValidationResultDispatcher>();
 
       foreach (var validator in validatorsMatchingToControls)
         validator.DispatchValidationFailures(validationResult);
     }
 
+    private void DispatchToValidationFailureRepository (IBusinessObjectValidationResult validationResult, BocList bocListControl)
+    {
+      var validationFailureRepository = bocListControl.ValidationFailureRepository;
+      validationFailureRepository.ClearAllValidationFailures();
+
+      var rowObjects = bocListControl.Value?.Cast<IBusinessObject>();
```

Comment:
Why is the cast necessary? The Value property already returns an IReadOnlyList<IBusinessObject> ?